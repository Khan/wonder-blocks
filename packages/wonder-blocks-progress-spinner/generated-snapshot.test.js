// @flow
// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-progress-spinner
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";
import CircularSpinner from "./components/circular-spinner.js";

describe("wonder-blocks-progress-spinner", () => {
    it("example 1", () => {
        const {CircularSpinner} = require("./index.js");

        const example = <CircularSpinner />;
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const {StyleSheet} = require("aphrodite");
        const {Text, View} = require("@khanacademy/wonder-blocks-core");
        const Color = require("@khanacademy/wonder-blocks-color").default;

        const styles = StyleSheet.create({
            contents: {
                flexDirection: "row",
            },
            dark: {
                background: Color.darkBlue,
                color: Color.white,
            },
            block: {
                width: 154,
                height: 154,
                marginRight: 16,
                alignItems: "center",
                justifyContent: "center",
                textAlign: "center",
            },
            inline: {
                display: "inline",
            },
        });

        const example = (
            <View>
                <View style={styles.contents}>
                    <View style={styles.block}>
                        <CircularSpinner size="large" />
                        <Text>size: large (96px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="medium" />
                        <Text>size: medium (48px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="small" />
                        <Text>size: small (24px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="xsmall" />
                        <Text>size: xsmall (16px)</Text>
                    </View>
                </View>

                <View style={[styles.contents, styles.dark]}>
                    <View style={styles.block}>
                        <CircularSpinner size="large" light />
                        <Text>light, size: large (96px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="medium" light />
                        <Text>light, size: medium (48px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="small" light />
                        <Text>light, size: small (24px)</Text>
                    </View>
                    <View style={styles.block}>
                        <CircularSpinner size="xsmall" light />
                        <Text>light, size: xsmall (16px)</Text>
                    </View>
                </View>

                <View>
                    <Text>
                        Inline inside
                        <CircularSpinner
                            size="xsmall"
                            style={styles.inline}
                        />{" "}
                        some text.
                    </Text>
                </View>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
