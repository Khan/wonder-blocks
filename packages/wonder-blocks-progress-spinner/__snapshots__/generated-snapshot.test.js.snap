// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-progress-spinner example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <svg
    className=""
    height={96}
    style={
      Object {
        "animationDuration": "1.1s",
        "animationIterationCount": "infinite",
        "animationName": Object {
          "0%": Object {
            "transform": "rotate(0deg)",
          },
          "100%": Object {
            "transform": "rotate(360deg)",
          },
          "50%": Object {
            "transform": "rotate(180deg)",
          },
        },
        "animationTimingFunction": "linear",
      }
    }
    viewBox="0 0 96 96"
    width={96}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className=""
      d="M88.38 46.91a3.818 3.818 0 1 1 7.602 0h.006c.008.362.012.725.012 1.09 0 26.51-21.49 48-48 48S0 74.51 0 48 21.49 0 48 0c.365 0 .728.004 1.09.012v.005a3.818 3.818 0 1 1 0 7.602v.032c-.362-.01-.725-.015-1.09-.015C25.708 7.636 7.636 25.708 7.636 48c0 22.292 18.072 40.364 40.364 40.364 22.292 0 40.364-18.072 40.364-40.364 0-.365-.005-.728-.015-1.09h.032z"
      fill-rule="nonzero"
      style={
        Object {
          "fill": "rgba(33,36,44,0.16)",
        }
      }
    />
  </svg>
</div>
`;
